src/app/plugins/pubsub.plugin.ts

```diff
@@ -28,60 +32,51 @@ export class PubSubPlugin extends Plugin {
         const subs = response.data;
 
         this._subs = subs.map((subData: { name: string }) => {
-          return subData.name
-            .toLowerCase()
-            .split('-')
-            .join(' ');
+          return subData.name.toLowerCase();
         });
-        this._generateEmbedSubs();
       })
       .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    if (args === undefined || args.length === 0) {
-      args = [''];
-    }
+    const subIn = this._parseCommand(args || []);
 
-    if (args[0].includes('type') || args[0].includes('list')) {
+    if (subIn.includes('type') || subIn.includes('list')) {
       //Simply return the list of supported subs
+      this._generateEmbedSubs();
       message.reply(this._embedSubs);
       return;
     }
 
-    const subIn = this._parseCommand(args);
+    console.log(`${this._subs[0] === subIn}`);
 
     // checks if their sub was a sub, then if that sub is recognised
-    const subEntry = this._subs.find((sub: string) => sub === subIn) || '';
+    const subType = this._subs.find((sub: string) => sub === subIn) || '';
 
-    if (subIn == 'reboot') {
-      this.updateList();
-      return;
-    } else if (subEntry == '' && subIn != 'random' && subIn != '') {
+    if (subType === '' && subIn !== 'random' && subIn !== '') {
       message.reply('Sub not found.  Use "list" or "type" to get a list of possible subs');
       return;
     }
 
-    this.container.loggerService.debug(subEntry);
+    this.container.loggerService.debug(subType);
```

Comment:
saa

Reply:
I don't actually know where this came from XD.  someone put it in my cat plugin so I thought it needed to be their XD